src/javascript/campaign-url-builder/components/campaign-url.js

```diff
@@ -125,35 +125,42 @@ export default class CampaignUrl extends React.Component {
    * state is activated. If there's an error shortening the URL, an alert
    * is displayed.
    */
-  shortenUrl = async () => {
-    this.setState({isShorteningUrl: true});
+  shortenUrl = () => {
+    this.setState({isShorteningUrl: true})
 
-    try {
-      let shortUrl = await shortenUrl(this.props.url);
+    shortenUrl(this.props.url)
+    .then(shortUrl => {
       this.setState({
         isShorteningUrl: false,
         shortUrl: shortUrl,
         showShortUrl: true,
       });
+      // TODO(nathanwest): Only send the event if the API was hit. Don't send
+      // events for cache hits.
       gaSendEvent({
         category: 'Campaign URL',
         action: 'shorten',
-        label: '(not set)',
-      });
-    } catch (err) {
+        label: 'success'
+      })
+    })
+    .catch(err => {
       AlertDispatcher.addOnce({
         title: 'Oops, an error occurred trying to shorten the URL',
         message: err.message,
-      });
+      })
       this.setState({
         isShorteningUrl: false,
         shortUrl: null,
         showShortUrl: false,
-      });
-    }
+      })
+      gaSendEvent({
+        category: "Campaign URL",
+        action: "shorten",
```

Comment:
Nit: maybe change the action here from "shorten" to something error-related. That would be more consistent with how actions are used in the rest of the app. Also then you could just have the error message be the entire label.

Reply:
- I wanted the action to be the same as with success, so that analytics aggregation can show all shorten attempts, along with what # failed. If analytics can easily group shorten + shorten-failure events together, then I can do this.